File: 1283-reformat-date/1283-reformat-date.go
import "fmt"
import "strings"

func reformatDate(date string) string {
    MONTH_MAP := map[string]string {
        "Jan": "1", 
        "Feb": "2",
        "Mar": "3", 
        "Apr": "4", 
        "May": "5", 
        "Jun": "6", 
        "Jul": "7", 
        "Aug": "8", 
        "Sep": "9", 
        "Oct": "10", 
        "Nov": "11", 
        "Dec": "12",
    }
    date_str := strings.Split(date, " ")
    day := "0" + date_str[0][:len(date_str[0])-2]
    month := "0" + MONTH_MAP[date_str[1]]
    year := date_str[2]
    return fmt.Sprintf("%s-%s-%s", year, month[len(month)-2:], day[len(day)-2:])
}

File: 1283-reformat-date/1283-reformat-date.py
class Solution:
    MONTH_MAP = {
        "Jan": "01",
        "Feb": "02",
        "Mar": "03",
        "Apr": "04",
        "May": "05",
        "Jun": "06",
        "Jul": "07",
        "Aug": "08",
        "Sep": "09",
        "Oct": "10",
        "Nov": "11",
        "Dec": "12",
        }
    def addZero(self, n: str) -> str:
        return '0' + n if len(n) == 1 else n
    
    def reformatDate(self, date: str) -> str:
        day, month, year = list(date.split())
        day = day[:-2]
        month = self.MONTH_MAP[month]
        return f"{year}-{month}-{self.addZero(day)}"
        